tests/workbenches/conftest.py

```diff
@@ -0,0 +1,203 @@
+from __future__ import annotations
+
+from typing import Generator
+
+import pytest
+from pytest_testconfig import config as py_config
+
+
+from tests.workbenches.utils import get_username
+
+from kubernetes.dynamic import DynamicClient
+from kubernetes.dynamic.exceptions import ResourceNotFoundError
+
+from ocp_resources.namespace import Namespace
+from ocp_resources.persistent_volume_claim import PersistentVolumeClaim
+from ocp_resources.route import Route
+from ocp_resources.notebook import Notebook
+
+
+from utilities.constants import Labels
+from utilities import constants
+from utilities.constants import INTERNAL_IMAGE_REGISTRY_PATH
+
+
+@pytest.fixture(scope="function")
+def users_persistent_volume_claim(
+    request: pytest.FixtureRequest, unprivileged_model_namespace: Namespace, unprivileged_client: DynamicClient
+) -> Generator[PersistentVolumeClaim, None, None]:
+    with PersistentVolumeClaim(
+        client=unprivileged_client,
+        name=request.param["name"],
+        namespace=unprivileged_model_namespace.name,
+        label={constants.Labels.OpenDataHub.DASHBOARD: "true"},
+        accessmodes=PersistentVolumeClaim.AccessMode.RWO,
+        size="10Gi",
+        volume_mode=PersistentVolumeClaim.VolumeMode.FILE,
+    ) as pvc:
+        yield pvc
+
+
+@pytest.fixture(scope="function")
+def minimal_image() -> Generator[str, None, None]:
+    """Provides a full image name of a minimal workbench image"""
+    image_name = "jupyter-minimal-notebook" if py_config.get("distribution") == "upstream" else "s2i-minimal-notebook"
+    yield f"{INTERNAL_IMAGE_REGISTRY_PATH}/{py_config['applications_namespace']}/{image_name}:{'2024.2'}"
+
+
+@pytest.fixture(scope="function")
+def default_notebook(
+    request: pytest.FixtureRequest,
+    admin_client: DynamicClient,
+    minimal_image: str,
+) -> Generator[Notebook, None, None]:
+    """Returns a new Notebook CR for a given namespace, name, and image"""
+    namespace = request.param["namespace"]
+    name = request.param["name"]
+
+    # Set new Route url
+    route_name = "odh-dashboard" if py_config.get("distribution") == "upstream" else "rhods-dashboard"
+    route = Route(client=admin_client, name=route_name, namespace=py_config["applications_namespace"])
+    if not route.exists:
+        raise ResourceNotFoundError(f"Route {route.name} does not exist")
+
+    # Set the correct username
+    username = get_username(dyn_client=admin_client)
+
+    probe_config = {
```

Comment:
I'm wondering how much of what is in these two dicts you expect to be dynamic vs constant. If at all possible I'd remove the portions that can be considered constant across different notebooks to make this slightly less verbose. Otherwise LGTM.